substrate/bin/minimal/node/src/chain_spec.rs

```diff
@@ -30,37 +31,25 @@ fn props() -> Properties {
 }
 
 pub fn development_config() -> Result<ChainSpec, String> {
-	let wasm_binary = WASM_BINARY.ok_or_else(|| "Development wasm not available".to_string())?;
-	Ok(ChainSpec::from_genesis(
-		"Development",
-		"dev",
-		ChainType::Development,
-		move || testnet_genesis(wasm_binary),
-		vec![],
-		None,
-		None,
-		None,
-		Some(props()),
-		None,
-	))
+	Ok(ChainSpec::builder(WASM_BINARY.expect("Development wasm not available"), Default::default())
+		.with_name("Development")
+		.with_id("dev")
+		.with_chain_type(ChainType::Development)
+		.with_genesis_config_patch(testnet_genesis())
+		.with_properties(props())
+		.build())
 }
 
 /// Configure initial storage state for FRAME pallets.
-fn testnet_genesis(wasm_binary: &[u8]) -> RuntimeGenesisConfig {
+fn testnet_genesis() -> Value {
 	use frame::traits::Get;
 	use runtime::interface::{Balance, MinimumBalance};
 	let endowment = <MinimumBalance as Get<Balance>>::get().max(1) * 1000;
 	let balances = AccountKeyring::iter()
 		.map(|a| (a.to_account_id(), endowment))
 		.collect::<Vec<_>>();
-	RuntimeGenesisConfig {
-		system: SystemConfig {
-			// Add Wasm runtime to storage.
-			code: wasm_binary.to_vec(),
-			_config: Default::default(),
-		},
-		balances: BalancesConfig { balances },
-		sudo: SudoConfig { key: Some(AccountKeyring::Alice.to_account_id()) },
-		..Default::default()
-	}
+	json!({
+		"balances": BalancesConfig { balances },
+		"sudo": SudoConfig { key: Some(AccountKeyring::Alice.to_account_id()) },
+	})
```

Comment:
I no longer see a way to do this in more type-safe way without `json!`, which is very annoying. Even though generated `GenesisConfig` is JSON-serializable, so I constructed it explicitly and then tried to serialize, serialization failed for me with this non-descriptive error:
```
Error("number out of range", line: 0, column: 0)
```

And it comes from `pallet-balances` :disappointed:

Any recommendations?

Reply:
Well, `json!` also doesn't work with `u128::MAX / 2` balance, while `ChainSpec::from_genesis` worked. Created an issue about this: https://github.com/paritytech/polkadot-sdk/issues/2963

Reply:
Can you try `arbitrary_precision` feature on `serde_json` crate?

Reply:
Please see the proposal of fix:
https://github.com/michalkucharczyk/polkadot-sdk/tree/mku-u128-balances-fix

`./minimal-node build-spec --chain dev` works fine (also `--raw`):
```
...
            [
              "5Fxune7f71ZbpP2FoY3mhYcmM596Erhv1gRue4nsPwkxMR4n",
              170141183460469231731687303715884105727
            ],
            [
              "5CUjxa4wVKMj3FqKdqAUf7zcEMr4MYAjXeWmUf44B41neLmJ",
              170141183460469231731687303715884105727
            ]
...
```

Reply:
If that works - great (it is a bit difficult to test due to a series of downstream patches on top of Substrate). Why did it work with `ChainSpec::from_genesis` though? It should have unified features of `serde_json` if one of the dependencies enabled `arbitrary_precision`.

Reply:
> (it is a bit difficult to test due to a series of downstream patches on top of Substrate

This single line in runtime/Cargo.tom should do the job:
```
serde_json = { version = "1.0.108", default-features = false, features = ["alloc", "arbitrary_precision"] }
```